pkg/pillar/containerd/oci_test.go

```diff
@@ -34,7 +34,7 @@ const loaderRuntimeSpec = `
             "/init.sh"
         ],
         "env": [
-            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
```

Comment:
Is this change required? If so is it required to quote all values e.g.,
FOO=1
needs to be changed to
FOO="1"

In any case, can't think break existing cloud-init/user-data for deployed containers?

Reply:
> In any case, can't think break existing cloud-init/user-data for deployed containers?

This is very valid concern: nothing can be worse than breaking compatibility with existing user data. 
To be on the safe side I can propose a small improvement to the existent change: here in this [line](https://github.com/lf-edge/eve/blob/7ad0126475f6ea698166ba4fbf4c1c3889d2554a/pkg/pillar/cmd/domainmgr/domainmgr.go#L1524) we compare old cloud-init version and a new one. The idea is that we can apply a new parsing mechanism only if versions differ. This will require a bit more of coding exercises (for example moving old and new versions retrieving  from the `doActivate` to the `configToStatus`), but that should not be a big deal. 

Also I would like @europaul to be involved in the review process (at least, Paul, please be aware of proposed changes, since you were the last one who touched the environment parsing logic).

Reply:
I took the test data from envparse's own [tests](https://github.com/hashicorp/go-envparse/blob/main/envparse_test.go) and ran them through the old parsing code + a `sh` process to see if we get any "false positives" - inputs that would cause an error when parsed by `envparse` but would not break the shell and the old parsing. This was the result:
```bash
"a b c=x"
"FOO=\"\\a\""
"FOO=\"\\"
"FOO=\"\\u12\""
"FOO=\"\\uabcZ\""
"FOO=\"abc \\uD83D\""
"FOO=\"abc \\uD83D \\uDE01\""
"FOO=\"abc \\uD83DDE01\""
"FOO=\"abc \\uD83D\\uDE0\""
```

That means that backward compatibility is broken on those kinds of env inputs. Even if we discard the weird unicode cases, the first three lines are still valid.

Here is the [test code](https://go.dev/play/p/DyTtRx1W0qT). It's not identical to the old and new versions, so I will run some manual tests with the full code base.

Reply:
The manual tests showed that the inputs
```
"a b c=x"
"FOO=\"\\"
"FOO=\"\\u12\""
```
are accepted by the old version. The others are caught fine by the change introduced in https://github.com/lf-edge/eve/pull/3752

Reply:
Thanks @europaul for the tests. Indeed, we cannot complain the new parser it's 100% backward compatible with the old one. However, the tests you mentioned are weird and should not be seen in a regular use case. In other words, they are correctly treated as wrong declarations by go-envparse, so let's keep the new parser (already merged) as it is... Nevertheless, it's really valid raise these points, thanks again.